vendor/parser-extended/parser.mly

```diff
@@ -4591,6 +4591,10 @@ atomic_type:
       { Ptyp_var (name, jkind) }
     | LPAREN mkrhs(UNDERSCORE {None}) COLON jkind=jkind_annotation RPAREN
       { Ptyp_var ($2, jkind) }
+    | LPAREN TYPE COLON jkind=jkind RPAREN
+      { if Erase_jane_syntax.should_erase ()
+        then Ptyp_any
+        else Ptyp_of_kind jkind }
```

Comment:
I don't think I agree here. As in, I don't think this new syntax is eraseable. But it really should be used only in contexts that _are_ erased, so it's probably a moot point. But I think I'd remove this, which could cause inscrutable errors.

Should we have some check in the compiler when seeing the new syntax around upstream compatibility? Probably, but it would be pretty subtle, because any use of this in a kind is fine (because the whole kind gets erased). I think I'd rather remove this bit of code and not check for erasability in the compiler. If someone does the bad thing (unlikely), the upstream build will break in an obvious way, which is fine.

Feel free to disagree. I'm pretty low confidence on most of this.